### PR TITLE
fix: when source code is not mounted resolves #45

### LINKED ISSUE #45: Unable to launch tutor using tutor-forum 18.1.0: empty volume list in docker compose files
## Issue

Enabling the tutor-forum plugin `forum` on tutor leverages tutor-forum `18.1.0`. Tutor is unable to launch with tutor-forum `18.1.0` due to blank `volume` attributes in the docker and docker.jobs docker compose files.

This is using tutor `18.1.3`, installed following the quickstart guide for new users.

Other details:
- Ubuntu 24.04
- amd64 arch

## Error
```
validating /home/user/.local/share/tutor/env/local/docker-compose.yml: services.forum.volumes must be a list
Error: Command failed with status 15: docker compose -f /home/user/.local/share/tutor/env/local/docker-compose.yml -f /home/user/.local/share/tutor/env/local/docker-compose.prod.yml --project-name tutor_local -f /home/user/.local/share/tutor/env/local/docker-compose.jobs.yml run --rm codejail-apparmor-job sh -e -c if [  "$SKIP_INIT" == "True" ]; then
    echo "Skipped AppArmor initialization"
else
    /usr/bin/loader -logtostderr -v=2 /profiles
fi
```

Looking the docker files generated by `tutor local launch` we can see:

`docker-compose.jobs.yml`:
```docker
    forum-job:
      image: docker.io/overhangio/openedx-forum:18.1.0
      environment:
        SEARCH_SERVER: "http://elasticsearch:9200"
        MONGODB_AUTH: ""
        MONGODB_HOST: "mongodb"
        MONGODB_PORT: "27017"
        MONGODB_DATABASE: "cs_comments_service"
        MONGOID_AUTH_SOURCE: "admin"
        MONGOID_AUTH_MECH: ""
        MONGOID_USE_SSL: "false"
      volumes:
      depends_on: ["elasticsearch", "mongodb"](tutor_env)
```

`docker-compose.yml`:
```docker
forum:
    image: docker.io/overhangio/openedx-forum:18.1.0
    environment:
      SEARCH_SERVER: "http://elasticsearch:9200"
      MONGODB_AUTH: ""
      MONGODB_HOST: "mongodb"
      MONGODB_PORT: "27017"
      MONGODB_DATABASE: "cs_comments_service"
      MONGOID_AUTH_SOURCE: "admin"
      MONGOID_AUTH_MECH: ""
      MONGOID_USE_SSL: "false"
    volumes:
    restart: unless-stopped
    depends_on: ["elasticsearch", "mongodb"]
 ```

Please note the empty `volumes:` attributes. These do not exist with `tutor-forum` versions before 18.1.0 (including 17.x). However, even if the user removes the empty `volumes:` attributes from the `forum` section of the docker files, `tutor local launch` will ensure the empty `volumes:` attributes are re-created and tutor will not be able to launch/init. It will show the error listed above.

## Expected behaviour
Tutor installed as per the Tutor quickstart guide, with the forum plugin enabled, works with no additional user input/customization. 

## Current workaround
Downgrade the `tutor-forum` plugin to `18.0.0` and everything works as expected using `pip install tutor-forum==18.0.0`.

### PR DESCRIPTION
**Pull request by ghassanmas**
Add a condition to test if mounted list does include entries, since empty `volume` defintion is not accepted in Docker compose syntax

### CODE REVIEW

File: changelog.d/20240828_131853_ghassan.maslamani_resolve_45.md [hunk 1]
```diff
@@ -0,0 +1 @@
+- [Bugfix] Fix when src is not mounted
```
Comment (DawoudSheraz) [inline changelog.d/20240828_131853_ghassan.maslamani_resolve_45.md:1]: nit: I think we can add more context to this changelog.

File: tutorforum/patches/local-docker-compose-jobs-services [hunk 2]
```diff
@@ -2,8 +2,10 @@ forum-job:
   image: {{ FORUM_DOCKER_IMAGE }}
   environment:
     {{ patch("forum-local-env")|indent(4) }}
+{%- if iter_mounts(MOUNTS, "forum")|first is defined %}
```
Comment (regisb) [inline tutorforum/patches/local-docker-compose-jobs-services:5]: In other plugins, we implement this with the following:

```
  {%- set mounts = iter_mounts(MOUNTS, "forum")|list %}
  {%- if mounts %}
  ...
  {%- endif %}
```

I suggest we do the same here.

**Review by regisb — COMMENTED** [review]
Sorry about breaking the build. Can we please merge this today and make a new release such that the forum plugin is functional once again?

**Comment by ghassanmas** [comment]
No worries ince it's reviewed again I will prepare a release

**Comment by ghassanmas** [comment]
do you any opposition to get this merged and released @regisb @DawoudSheraz

**Review by regisb — APPROVED** [review]
LGTM! Let's merge this and make a new release ASAP!

**Comment by ghassanmas** [comment]
Here  #47 we have it
